MultiLangProgGuide/MultiLangProgGuide.rst

```diff
@@ -2717,23 +2771,133 @@ Go Comprehensive API
+
+    YDBExit() int
+
+If a process wishes to close YottaDB databases and no longer use
+YottaDB, a call to :code:`YDBExit()` wraps `ydb_exit()`_ so that any
+further calls to YottaDB result in a :code:`C.YDB_ERR_CALLINAFTEREXIT`
+error.
+
+Typical processes will not need to call :code:`YDBExit()` because
+normal process termination closes databases cleanly. However, a
+process that has completed its use of YottaDB, but intends to continue
+with other work for some material period of time, should call
+:code:`YDBExit()`.
+
+-----------------
+Go YDBForkNCore()
+-----------------
+
+.. code-block:: go
+
+    YDBForkNCore()
+
+The function wraps `ydb_fork_n_core()`_ to generate a core file
+snapshot of a process for debugging purposes. See `ydb_fork_n_core()`_
+for more information.
 
-GetSubBufAlloc(n)
-GetSubBufUsed(n)
+:code:`YDBForkNCore()` has no parameters and returns no return code.
+
+------------
+Go YDBFree()
+------------
+
+.. code-block:: go
+
+    YDBFree(ptr unsafe.pointer)
+
+The function wraps `ydb_free()`_ to release memory previously
+allocated using :code:`YDBMalloc()`. As passing a :code:`ptr` not
+previously allocated using :code:`YDBMalloc()` will result in
+unpredictable behavior, application code should be written with an
+appropriate level of diligence when calling :code:`YDBFree()`.
+
+------------
+Go YDBInit()
+------------
+
+.. code-block:: go
+
+    YDBInit() int
+
+The function wraps `ydb_init()`_ to initialize the YottaDB runtime
+system. This call is normally not required as YottaDB initializes
+itself on its first call, the exception being when an application
+wishes to set its own signal handlers (see
+`Signals`_). :code:`YDBInit()` returns :code:`C.YDB_OK` or an `error
+return code`_.
+
+--------------
+Go YDBMalloc()
+--------------
+
+.. code-block:: go
+
+    YDBMalloc(size uint64) unsafe.pointer
+
+The function wraps `ydb_malloc()`_ to allocate :code:`size` bytes of
+storage managed by YottaDB. Use of :code:`YDBMalloc()` to allocate
+storage provides debugging tools. Using any function other than
+:code:`YDBFree()` to release storage allocated with
+:code:`YDBMalloc()` has unpredictable results.
+
+As the definition and implementation of Go protect against dangling
+pointers in pure Go code, Go application code may not be designed and
+coded with the same level of defensiveness against dangling pointers
+that C applications are. Please take additional care in designing and
+coding your application to ensure the correct use of application
+storage allocated using :code:`YDBMalloc()`.
 
 --------------------------
-Go BufferT Utility Methods
+Go YDBStdoutStdErrAdjust()
 --------------------------
 
---------------------
-Go Utility Functions
---------------------
+.. code-block:: go
+
+    YDBStdoutStderrAdjust() int
+
+Refer to the description of `ydb_stdout_stderr_adjust()`_, which this
+function wraps. The return code is :code:`C.YDB_OK`.
+
+Applications that do not mix M and non-M code do not need this
```

Comment:
Is this true? I thought it was if stdout and stderr were being used by various components that one might want to use this routine. Use of this routine from Golang is currently an unknown. We don't know how Golang sets up its IO stuff. Would be interesting to see if using this in Golang then forking off a process caused any issues.